doc/advanced-configuration/outgoing-connections.md

```diff
@@ -188,278 +103,143 @@ sslmode     = verify-full
 
-Currently only one Riak connection pool can exist for each supported XMPP host.
-It is configured with the following tuple inside the `outgoing_pools` config option.
+## Riak options
 
-```erlang
-{outgoing_pools, [
- {riak, global, default, [{workers, 20}], [{address, "127.0.0.1"}, {port, 8087}]}
-]}.
-```
+Currently only one Riak connection pool can exist for each supported XMPP host (the default pool).
 
 *WARNING:* `riak` backend is not compatible with `available_worker` strategy.
 
-#### Riak SSL connection setup
-
-Using SSL for Riak connection requires passing extra options in `ConnectionOptions` to the
-aforementioned `riak` tuple.
+#### `outgoing_pools.riak.*.connection.address`
+* **Syntax:** string
+* **Example:** `address = "127.0.0.1"`
 
-Here is the proper syntax:
+#### `outgoing_pools.riak.*.connection.port`
+* **Syntax:** integer
+* **Example:** `port = 8087`
 
-* **Credentials**
-    * **Description:** Specifies credentials to use to connect to the database.
-    * **Syntax:** `{credentials, User, Password}`
-    * **Supported values** `User` and `Password` are strings with a database username and password respectively.
+#### `outgoing_pools.riak.*.connection.credentials`
+* **Syntax:** `{user = "username", password = "pass}`
+* **Default:** none
+* **Example:** `credentials = {user = "myuser", password = "tisismepasswd"}`
+* **Comment:** optional - setting this option forces connection over TLS
 
-* **CACert**
-    * **Description:** Specifies a path to the CA certificate that was used to sign the database certificates.
-    * **Syntax:** `{cacertfile, Path}`
-    * **Supported values** `Path` is a string with a path to the CA certificate file.
+Riak supports also all TLS-specific options described in TLS section.
 
-* **SSL_Opts**
-    * **Description**: list of SSL options as defined in [Erlang SSL module doc](http://erlang.org/doc/man/ssl.html)
-      They will be passed verbatim to the `ssl:connect` function.
-    * **Syntax** `{ssl_opts, ListOfSSLOpts}`
-    * **Example**:
+## Cassandra options
 
-```erlang
-{ssl_opts, [{ciphers, ["AES256-SHA", "DHE-RSA-AES128-SHA256"]},
-            {server_name_indication, disable}]}
-```
+#### `outgoing_pools.cassandra.*.connection.servers`
+* **Syntax:** a list of maps containing keys `"ip_adddress"` and `"port"`
+* **Default:** `[{ip_address = "localhost", port = 9042}]`
+* **Example:** `servers = [{ip_address = "host_one", port = 9042}, {ip_address = "host_two", port = 9042}]`
 
-##### Example configuration
+#### `outgoing_pools.cassandra.*.connection.keyspace`
+* **Syntax:** string
+* **Default:** `"mongooseim"`
+* **Example:** `keyspace = "big_mongooseim_database"`
 
-An example configuration can look as follows:
+#### `outgoing_pools.cassandra.*.connection.auth`
+* **Syntax:** {module = string, options = list of pairs of strings}
+* **Default:** undefined (no auth)
+* **Example:** `auth = {module = "cqerl_auth_plain_handler", options = [["test", "aaa"]]}`
```

Comment:
Yes, and this is the part I'm not sure about. It is a nested list so that you can pass multiple entries - while parsing it is changed into a proplist. The plain hander expects only one pair and it denotes username and password, but you can put a custom module here and you may want to set your custom configuration here - while in current state of our TOML parser it is not possible, it accepts only predefined values. So, I came up with a crude way of passing arbitrary values, which your custom cassandra auth module have to interpret, because they are always pairs of binaries. Ugly as hell, I admit. The only alternative I can think of is saying: you can use only plain handler, pass username and password, end of story.

Reply:
I think it should be done differently:
- the formt should be: `auth = {module = "cqerl_auth_plain_handler", options.test = "aaa"}`
- the parser should use `parse_section` for `options`

Reply:
If we want to be strict, perhaps we should choose to support only plain handler. I'm getting more and more inclined to do so. Any other solution would be confusing and would allow too much (that's actually the whole point of it).